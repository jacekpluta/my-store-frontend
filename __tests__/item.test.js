import React from "react";
import Item from "../components/Item";
import { shallow } from "enzyme";

let wrapper;

const fakeItem = {
  description: "test",
  id: "123abc",
  image: "testImage",
  largeImage: "testLargeImage",
  price: 111,
  title: "test",
};

beforeEach(() => {
  wrapper = shallow(<Item item={fakeItem} />);
});

it("renders and maches snapshot", () => {
  const wrapper = shallow(<Item item={fakeItem} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders price tag", () => {
  const PriceTag = wrapper.find("PriceTag");
  expect(PriceTag.children().text()).toBe("$1.11");
  expect(wrapper.find("Title a").text()).toBe(fakeItem.title);

  const img = wrapper.find("img");
  expect(img.props().src).toBe(fakeItem.image);
  expect(img.props().alt).toBe(fakeItem.title);
});

it("renders title", () => {
  expect(wrapper.find("Title a").text()).toBe(fakeItem.title);
});

it("renders image", () => {
  const img = wrapper.find("img");
  expect(img.props().src).toBe(fakeItem.image);
  expect(img.props().alt).toBe(fakeItem.title);
});

it("renders out buttons", () => {
  const buttonList = wrapper.find(".buttonList");
  expect(buttonList.children()).toHaveLength(3);

  expect(buttonList.find("Link")).toHaveLength(1);
  expect(buttonList.find("addToCart")).toHaveLength(1);
  expect(buttonList.find("deleteItem")).toHaveLength(1);
});
