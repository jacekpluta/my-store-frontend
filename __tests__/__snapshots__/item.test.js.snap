// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and maches snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="test"
    src="testImage"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "123abc",
          },
        }
      }
    >
      <a>
        test
      </a>
    </Link>
  </Title>
  <PriceTag>
    $1.11
  </PriceTag>
  <p>
    test
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "123abc",
          },
        }
      }
    >
      <a>
        Edit 
      </a>
    </Link>
    <addToCart
      itemId="123abc"
    />
    <deleteItem
      itemId="123abc"
    >
      Delete Item
    </deleteItem>
  </div>
</ItemStyles__Item>
`;
